File: backend/src/main/java/com/awais/hr/module/attendance/service/AttendanceService.java
package com.awais.hr.module.attendance.service;

import com.awais.hr.module.attendance.dto.CheckInRequestDTO;
import java.util.List;
import java.util.Map;

public interface AttendanceService {
    List<Map<String, Object>> getAttendanceHistory(String email);
    void checkIn(String email, CheckInRequestDTO dto, String ipAddress);
    void checkOut(String email);
    void deleteAttendance(String id);
}

File: frontend/src/services/attendanceService.js
import apiClient from './api';

export const getAttendanceHistory = () => {
  return apiClient.get('/attendance');
};

export const checkIn = (dto) => {
  return apiClient.post('/attendance/checkin', dto);
};

export const checkOut = () => {
  return apiClient.post('/attendance/checkout');
};

export const deleteAttendance = (id) => {
  return apiClient.delete(`/attendance/${id}`);
};
